### PR TITLE
small hydration improvements

### PR DESCRIPTION
**Pull request by fa-sharp**
This implements the small changes discussed in #46.

If the `onInit` parameter passed to the hook is a function, it is run to reset the render value when the storage unmounts. This allows the advanced use case of handling the loading/hydration state in the `onInit` function.

I also renamed the boolean parameter to `isHydrated`, which better describes the state it represents. This is purely a documentation change.

### CODE REVIEW

File: src/hook.ts
Comment (louisgv) [inline src/hook.ts:95]: nit: let's put this under block inside curly brackets to keep it consistent with the repo

**Review by louisgv — APPROVED** [review]
Beside the nits, LGTM! Once it's fixed, I will merge and cut a new release likely next week?

**Comment by fa-sharp** [comment]
Fixed formatting! Let me know if there's anything else. Thanks!!

**Pull request merged**
